code/modules/reagents/chemistry_reagents/stims.dm

```diff
@@ -0,0 +1,164 @@
+
+/datum/reagent/stimulant
+	reagent_state = LIQUID
+	custom_metabolism = AMOUNT_PER_TIME(1, 1 MINUTES) // Consumes 1 unit per minute.
+	overdose = LOWH_REAGENTS_OVERDOSE
+	overdose_critical = LOWH_REAGENTS_OVERDOSE_CRITICAL
+	chemclass = CHEM_CLASS_NONE
+	flags = REAGENT_SCANNABLE | REAGENT_TYPE_STIMULANT
+	var/jitter_speed = 0.3 SECONDS
+	var/jitter_per_amount = 2
+	var/jitter = 2
+
+/datum/reagent/stimulant/on_mob_life(mob/living/M, alien, delta_time)
+	. = ..()
+	// Stimulants drain faster for each stimulant in the drug.
+	// Having 2 stimulants means the duration will be 2x shorter, having 3 will be 3x shorter, etc
+	if(holder)
+		for(var/i in holder.reagent_list)
+			if(i == src)
+				continue
+
+			var/datum/reagent/R = i
```

Comment:
```
for(var/datum/reagent/reagent as anything in holder.reagent_list)
```

(var name, loop var, etc)

Reply:
Git blame Spooky - https://github.com/cmss13-devs/cmss13/blame/f95e86ecb6bc70d7c0b62d72a472ca6daa9ffe03/code/modules/cm_tech/techs/marine/tier2/combat_stimulants.dm . I'll fix it up!